### ISSUE TITLE
backup: restored data has a different number of rows

### ISSUE THREAD

**Issue by danhhz** [cockroachdb/cockroach#15618]
The backup/restore production testing recently had a row count mismatch. The restored data had two fewer rows that the data selected for backup https://teamcity.cockroachdb.com/viewLog.html?buildId=240534&buildTypeId=Internal_Production_BackupToAzure&tab=buildLog#_

I'm still investigating if this is a bug in backup/restore or in the production testing script or something else

**Comment by danhhz** [comment]
Important to note: the full backup worked fine, this is limited to the incremental backup

**Comment by danhhz** [comment]
I tried another restore of the same data and got the same resulting row count on the restored data 117603747. Also modified dump to skip the block_writer payload and am dumping the contents of the two tables. The latter will take quite a while, but hopefully will finish overnight

**Comment by danhhz** [comment]
Reran exactly the same incremental backup as the test and it's slightly larger (305440392 vs 305440735). Started a restore of this data to see how many rows are in it. Will check on this and dump in the morning.

**Comment by danhhz** [comment]
I had a theory about stale reads this morning and discovered (with @spencerkimball's help) that Export was missing the updatesTSCache flag. Yesterday was also the first day we'd had some clock offset issues on lapis, so this is all consistent with missing two rows in a backup. The flag fix will merge before rc2 is cut.

Luckily, the second incremental backup I ran at the same timestamp should have enough info to confirm this, but I'll have to write some code to confirm. Doing that now

**Comment by danhhz** [comment]
The second incremental backup (should have been exactly the same) indeed had two rows not present in the original.

```
/Table/51/1/2743393367451462824/"c382a6bd-1ed7-43e4-94c8-222ba5daaade"/479649/0/1493746724.058476569,1 /TUPLE/4:4:Bytes/[...]
/Table/51/1/2743558694436043066/"d62f4bfb-ddb6-446f-a679-4cf345f8bf5e"/476164/0/1493746724.058476569,1 /TUPLE/4:4:Bytes/[...]
```

The mvcc timestamps on them is less than a second before the timestamp of the incremental backup. This is consistent with the theory that a write came along after the Export with a lower timestamp. Optimistically assuming this is fixed by #15642
